MCNPInput.cpp

```diff
@@ -1273,16 +1275,34 @@ void InputDeck::parseSurfaces( LineExtractor& lines ){
     if( do_line_continuation( lines, token_buffer ) ){
       continue;
     }
-   
-    SurfaceCard* s = new SurfaceCard(*this, token_buffer);
-
-    if( OPT_VERBOSE) s->print(std::cout);
+    int numFacets = 0;
+    
+    if(token_buffer.at(1).find_first_of("1234567890-") != 0){
+      mnemonic = token_buffer.at(1);
+    }
+    else{
+      mnemonic = token_buffer.at(2);
+    }
+    
+    if( mnemonic == "box" || mnemonic == "rpp" ){
```

Comment:
A little cleaner to implement this in a function that returns numFacets for a given mnemonic.

That function could 
* declare a std::map< std::string, int> at local scope:  mnemonic -> numFacets
* search the keys of that map for the mnemonic
* if the key is found - return numFacets
* else return 0

This is extensible by just adding to the map and done.